### DISCUSSION TITLE
What is the "Voice" in AlphaTab's data model hierachy?

### DISCUSSION THREAD

**Discussion by lannka** [CoderLine/alphaTab#585] [Q&A] [answered]
> A Voice allows to have multiple independent notes placed within the same bar indicating the simultaneous playing of them with different duration.
> The voice is again just a container for the individual beats played in it.

To help me understand, how "Voice" is different from having a separate "Track"? Could you share a real world example how it can be useful?

BTW, I think there is a mistake in the [doc](https://www.alphatab.net/docs/reference/score):
<img width="450" alt="Screen Shot 2021-04-04 at 8 49 26 PM" src="https://user-images.githubusercontent.com/8496897/113535075-483ccd80-9587-11eb-9e29-cd0c23093e3c.png">
There should be "Bars" in between "Staves" & "Voices"

**Comment by lannka** [comment]
I figured out. It's a standard music term to describe multiple notes played together. For example a chord.

The doc still needs to be fixed though.

**Reply by Danielku15** [reply]
Hi there. 
Sorry for the delayed response. I once started writing an answer and thought I actually sent it but it seems I missed it. I think your interpretation is not fully correct. Within alphaTab a bunch of notes played together is a **Beat**. Voices you can treat as multiple "layers" of notes that are played within the same bar. It's like multiple staves squashed into one. Each voice has an own rhythm (e..g first voice has quarter notes and second one has half notes). 

Here some references on that topic:
https://musescore.org/en/handbook/3/voices
https://music.stackexchange.com/questions/22678/notation-of-multiple-voices-in-piano-music
http://lilypond.org/doc/v2.18/Documentation/notation/multiple-voices
https://music.stackexchange.com/questions/39608/can-someone-explain-what-a-voice-is-in-piano

**Comment by Danielku15** [comment, marked as answer]
Docs are updated with some visual examples. I hope this way it becomes more clear what a "voice" is. 

https://www.alphatab.net/docs/reference/score/

**Reply by lannka** [reply]
The visualization is awesome! Thank you.
